libs/datamodel/core/tests/attributes/relations/referential_actions.rs

```diff
@@ -130,11 +129,80 @@ fn actions_on_prisma_referential_integrity() {
         parse(&dml)
             .assert_has_model("B")
             .assert_has_relation_field("a")
-            .assert_relation_delete_strategy(*action)
-            .assert_relation_update_strategy(*action);
+            .assert_relation_delete_strategy(*action);
     }
 }
 
+#[test]
+fn on_update_actions_should_not_work_on_prisma_referential_integrity() {
+    let dml = indoc! { r#"
+        datasource db {
+          provider = "mysql"
+          url = "mysql://"
+          referentialIntegrity = "prisma"
+        }
+
+        generator client {
+          provider = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        model A {
+          id Int @id
+          bs B[]
+        }
+
+        model B {
+          id Int @id
+          aId Int
+          a A @relation(fields: [aId], references: [id], onUpdate: Restrict)
+        }
+    "#};
+
+    let expect = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@relation": Referential actions other than `NoAction` will not work for `onUpdate` without foreign keys. Please follow the issue: https://github.com/prisma/prisma/issues/9014[0m
+          [1;94m-->[0m  [4mschema.prisma:20[0m
+        [1;94m   | [0m
+        [1;94m19 | [0m  aId Int
+        [1;94m20 | [0m  a A @relation(fields: [aId], references: [id], [1;91monUpdate: Restrict[0m)
+        [1;94m   | [0m
+    "#]];
+
+    expect.assert_eq(&datamodel::parse_schema(dml).map(drop).unwrap_err());
+}
+
+#[test]
+fn on_update_no_action_should_work_on_prisma_referential_integrity() {
+    let dml = indoc! { r#"
+        datasource db {
+          provider = "mysql"
+          url = "mysql://"
+          referentialIntegrity = "prisma"
+        }
+
+        generator client {
+          provider = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        model A {
+          id Int @id
+          bs B[]
+        }
+
+        model B {
+          id Int @id
+          aId Int
+          a A @relation(fields: [aId], references: [id], onUpdate: NoAction)
+        }
+    "#};
+
+    parse(&dml)
+        .assert_has_model("B")
+        .assert_has_relation_field("a")
+        .assert_relation_update_strategy(ReferentialAction::NoAction);
```

Comment:
I like how readable this testing helper is. Nice improvement over matching strings with color codes embedded.

Reply:
The snapshot tests are ugly to read, but brilliant when you have test failures! You get them printed what is expected, actual and diff, all in color. And, if you accept the changes, you just `env UPDATE_EXPECT=1 cargo test` and the expected strings will update automatically to what it actually is.

Reply:
We have countless of tests matching large string blobs, updating them manually when we have one extra word added is not fun, and it's quite cool to just have one command to update them all automatically.

Reply:
![2021-09-15_20-09-1631730407](https://user-images.githubusercontent.com/34967/133488865-be851674-3d59-4ddd-a67c-3f5d44abc504.png)

See this? It is important we test what parts are highlighted from the errors. We have a problem, if they are wrong.

Reply:
The embedded error codes look bad on github and when editing source, but great when they fail. It's a tradeoff. Also the assertion helpers test successful parses, while we use snapshot tests for errors, so they don't really overlap. In general, I'm finding more and more places where snapshot tests are just better than structured assertions and I think @pimeys agrees.

Reply:
Yep. Taking a step back how the snapshot tests look like for errors with color coding, it has helped us tremendously with an absolutely rigorous task of maintaining them when we change how the errors look like...